src/sql/workbench/contrib/notebook/test/electron-browser/cell.test.ts

```diff
@@ -1234,6 +1234,28 @@ suite('Cell Model', function (): void {
 		assert.deepStrictEqual(cellModel.attachments, attachments, 'addAttachment should not add duplicate images');
 	});
 
+	test('deleting attachment from cell source should remove it from attachments should add a valid attachment to cell', async function () {
+		let imageFilebase64Value = 'data:application/octet-stream;base64,iVBORw0KGgoAAAANSU';
+		let index = imageFilebase64Value.indexOf('base64,');
+		const testImageAttachment: nb.ICellAttachment = { ['image/png']: imageFilebase64Value.substring(index + 7) };
+		let attachments: nb.ICellAttachments = { 'test.png': testImageAttachment };
+		let notebookModel = new NotebookModelStub({
+			name: '',
+			version: '',
+			mimetype: ''
+		});
+		let contents: nb.ICellContents = {
+			cell_type: CellTypes.Markdown,
+			source: '',
+			metadata: {}
+		};
+		let model = factory.createCell(contents, { notebook: notebookModel, isTrusted: false });
+		model.addAttachment('image/png', imageFilebase64Value, 'test.png');
+		assert.deepStrictEqual(model.attachments, attachments, 'attachment was not added initially');
+		model.source = '';
```

Comment:
Just for sanity, let's set the source to something that isn't an empty string first. Would love to exercise the regex a bit in our unit tests.

Reply:
Yeah good call. I added some more cases, let me know if you think of others.

I thought about refactoring out the cleanup function to be unit testable by itself but because this is only being used in this one specific place that didn't seem useful, I'd rather we just make sure we have test coverage at the cell level for all scenarios so we can be sure that all the properties are updated as expected as well. (if there's a bunch more to add we can look into consolidating the logic for the tests to reduce the overhead there)

Reply:
Appreciate it, thanks. I think you've covered the main scenarios here, so I'm happy with these tests for now.

Agreed that refactoring out the cleanup function isn't necessary in this case.